docs/DeveloperGuide.md

```diff
@@ -5,9 +5,238 @@
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
```

Comment:
For all your images, save it as a proper file inside your docs folder. Avoid copy pasting and letting Github render from blobs. This will help in version control as well.